### PR TITLE
[REF] Add in LineItem APIv4 Entity

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
Add in the Line Item entity to APIv4

Before
----------------------------------------
LineItem not present in APIv4

After
----------------------------------------
LineItem present in APIv4

ping @eileenmcnaughton @colemanw

### CODE REVIEW

File: Civi/Api4/Action/LineItem/Create.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ * $Id$
```
Comment (eileenmcnaughton) [inline Civi/Api4/Action/LineItem/Create.php]: we don't need ids

**Comment by eileenmcnaughton** [comment]
@seamuslee001 what is the thinking behind bringing check_tax_amount across - it looks a look like apiv3  baggage / cruft

**Comment by seamuslee001** [comment]
@eileenmcnaughton i wasn't sure what to do with that it seemed like it was serving a purpose and made more sense to be in the centralised BAO Create. It seemed like the only times that BAO::create was being called was to get away from the check tax amount function

**Comment by eileenmcnaughton** [comment]
@seamuslee001 I think I'd leave it out  - the function is so unloved it has things like this in it

```
            if ($params[$field] != $params['prevContribution']->$field) {
            }
```

LineItem.create is  currently heavily bypassed so I wouldn't assume the v3 api  is better than the BAO

File: Civi/Api4/Action/LineItem/Create.php (hunk 1 above)
Comment (colemanw) [reply Civi/Api4/Action/LineItem/Create.php]: Where did that bit o' boilerplate come from?

**Comment by seamuslee001** [comment]
ok I have made the changes now @eileenmcnaughton @colemanw

File: Civi/Api4/LineItem.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+
+namespace Civi\Api4;
+
+/**
+ * LineItem Entity.
+ * @package Civi\Api4
+ */
```
Comment (colemanw) [inline Civi/Api4/LineItem.php]: This PR looks good to me code-wise.

The only thing missing is some documentation; in APIv4 I've been trying to add a sentence or two in the docblock for each entity saying what an entity is/does and how/when to use it, plus a `@see` annotation linking to relevant documentation.

**Comment by eileenmcnaughton** [comment]
I wonder if financial type acls are working on this api

**Comment by mattwire** [comment]
@seamuslee001 There's a question from @eileenmcnaughton re financial type acls

**Comment by eileenmcnaughton** [comment]
This is stale - closing for now

**Pull request closed without merging**